pool/pool.go

```diff
@@ -30,284 +31,576 @@ import (
-	NodeRequestTimeout        time.Duration
-	ClientRebalanceInterval   time.Duration
-	SessionTokenThreshold     time.Duration
-	SessionExpirationDuration uint64
-	nodesParams               []*NodesParam
-	clientBuilder             func(endpoint string) (Client, error)
-}
-
-type NodesParam struct {
+// client is a wrapper for sdkClient.Client to generate mock.
+type client interface {
+	BalanceGet(context.Context, sdkClient.PrmBalanceGet) (*sdkClient.ResBalanceGet, error)
+	ContainerPut(context.Context, sdkClient.PrmContainerPut) (*sdkClient.ResContainerPut, error)
+	ContainerGet(context.Context, sdkClient.PrmContainerGet) (*sdkClient.ResContainerGet, error)
+	ContainerList(context.Context, sdkClient.PrmContainerList) (*sdkClient.ResContainerList, error)
+	ContainerDelete(context.Context, sdkClient.PrmContainerDelete) (*sdkClient.ResContainerDelete, error)
+	ContainerEACL(context.Context, sdkClient.PrmContainerEACL) (*sdkClient.ResContainerEACL, error)
+	ContainerSetEACL(context.Context, sdkClient.PrmContainerSetEACL) (*sdkClient.ResContainerSetEACL, error)
+	EndpointInfo(context.Context, sdkClient.PrmEndpointInfo) (*sdkClient.ResEndpointInfo, error)
+	NetworkInfo(context.Context, sdkClient.PrmNetworkInfo) (*sdkClient.ResNetworkInfo, error)
+	ObjectPutInit(context.Context, sdkClient.PrmObjectPutInit) (*sdkClient.ObjectWriter, error)
+	ObjectDelete(context.Context, sdkClient.PrmObjectDelete) (*sdkClient.ResObjectDelete, error)
+	ObjectGetInit(context.Context, sdkClient.PrmObjectGet) (*sdkClient.ObjectReader, error)
+	ObjectHead(context.Context, sdkClient.PrmObjectHead) (*sdkClient.ResObjectHead, error)
+	ObjectRangeInit(context.Context, sdkClient.PrmObjectRange) (*sdkClient.ObjectRangeReader, error)
+	ObjectSearchInit(context.Context, sdkClient.PrmObjectSearch) (*sdkClient.ObjectListReader, error)
+	SessionCreate(context.Context, sdkClient.PrmSessionCreate) (*sdkClient.ResSessionCreate, error)
+}
+
+// InitParameters contains values used to initialize connection Pool.
+type InitParameters struct {
+	key                       *ecdsa.PrivateKey
+	logger                    *zap.Logger
+	nodeConnectionTimeout     time.Duration
+	nodeRequestTimeout        time.Duration
+	clientRebalanceInterval   time.Duration
+	sessionTokenThreshold     time.Duration
+	sessionExpirationDuration uint64
+	nodeParams                []NodeParam
+
+	clientBuilder func(endpoint string) (client, error)
+}
+
+// SetKey specifies default key to be used for the protocol communication by default.
+func (x *InitParameters) SetKey(key *ecdsa.PrivateKey) {
+	x.key = key
+}
+
+// SetLogger specifies logger.
+func (x *InitParameters) SetLogger(logger *zap.Logger) {
+	x.logger = logger
+}
+
+// SetNodeConnectionTimeout specifies the timeout for connection to be established.
+func (x *InitParameters) SetNodeConnectionTimeout(timeout time.Duration) {
+	x.nodeConnectionTimeout = timeout
+}
+
+// SetNodeRequestTimeout specifies the timeout for request to node to decide if it is alive.
```

Comment:
What do you think about `SetHealthcheckTimeout`? IMO it'd be more clear. Current name implies any request to the node.